uxarray/grid/grid.py

```diff
@@ -410,14 +409,24 @@ def Mesh2_node_cart_y(self) -> xr.DataArray:
     @property
     def Mesh2_face_y(self) -> xr.DataArray:
         """UGRID Coordinate Variable ``Mesh2_face_y``, which contains the
-        latitude of each face center.
+        latitude or cartesian coordinate of each face center.
 
         Dimensions (``nMesh2_face``)
         """
-        if "Mesh2_face_y" in self._ds:
-            return self._ds["Mesh2_face_y"]
-        else:
-            return None
+        if "Mesh2_face_y" not in self._ds:
+            _centroid_from_mean_verts(self)
+        return self._ds['Mesh2_face_y']
+
+    @property
+    def Mesh2_face_z(self) -> xr.DataArray:
```

Comment:
Now that I think of it do we actually want this since sometimes you won't have a face_z?

Reply:
Yeah, we don't need `Mesh2_face_z` since this has no meaning with lat/lon coords.

We will however have a `Mesh2_face_cart_z` that's getting implemented in #455